# Introducing Thanksgiving Dinner Data

In [28]:
import pandas as pd
data = pd.read_csv("thanksgiving.csv", encoding="Latin-1")
print(data.head())

   RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                            

# Display all of column names

In [3]:
print(data.columns)

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

# Filtering Out Rows From A DataFrame

In [31]:
print(data[ "Do you celebrate Thanksgiving?"].value_counts())
data = data[data["Do you celebrate Thanksgiving?"]=="Yes"]
print(data)

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64
      RespondentID Do you celebrate Thanksgiving?  \
0       4337954960                            Yes   
1       4337951949                            Yes   
2       4337935621                            Yes   
3       4337933040                            Yes   
4       4337931983                            Yes   
5       4337929779                            Yes   
6       4337924420                            Yes   
7       4337916002                            Yes   
8       4337914977                            Yes   
9       4337899817                            Yes   
11      4337893416                            Yes   
12      4337888291                            Yes   
13      4337878450                            Yes   
14      4337878351                            Yes   
15      4337857295                            Yes   
16      4337856362                            Yes   
17      4337854106        

# Using Value_counts To Explore Main Dishes


In [37]:
print(data["What is typically the main dish at your Thanksgiving dinner?"].value_counts())
data_Tofurkey=data[data["What is typically the main dish at your Thanksgiving dinner?"]=="Tofurkey"]
print(data_Tofurkey["Do you typically have gravy?"])

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64
4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object


# Figuring Out What Pies People Eat

In [39]:
print(data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple"].value_counts())
print(data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin"].value_counts())
print(data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan"].value_counts())

Apple    514
Name: Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple, dtype: int64
Pumpkin    729
Name: Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin, dtype: int64
Pecan    342
Name: Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan, dtype: int64


In [52]:
apple_isnull = data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple"].isnull()
pumpkin_isnull = data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin"].isnull()
pecan_isnull = data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan"].isnull()
ate_pies = apple_isnull & pumpkin_isnull & pecan_isnull
print(ate_pies.value_counts())
data[-ate_pies][["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple", "Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin", "Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan"]]

False    876
True     104
dtype: int64


Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple  \
0                                                 Apple                                                         
1                                                 Apple                                                         
2                                                 Apple                                                         
3                                                   NaN                                                         
4                                                 Apple                                                         
6                                                 Apple                                                         
8                                                 Apple                                                         
9                                                   NaN                                                         
11                                                Apple                                                         
12                                                  NaN                                                         
13                                                Apple                                                         
14                                                  NaN                                                         
16                                                  NaN                                                         
17                                                  NaN                                                         
18                                                Apple                                                         
19                                                Apple                                                         
20                                                  NaN                                                         
23                                                Apple                                                         
24                                                  NaN                                                         
25                                                Apple                                                         
26                                                Apple                                                         
27                                                Apple                                                         
28                                                Apple                                                         
29                                                Apple                                                         
30                                                  NaN                                                         
32                                                Apple                                                         
33                                                  NaN                                                         
34                                                Apple                                                         
36                                                  NaN                                                         
37                                                Apple                                                         
...                                                 ...                                                         
1019                                              Apple                                                         
1021                                                NaN                                                         
1022                                              Apple                                                         
1023                                              

# Converting Age To Numeric

In [118]:
def age_conversion(string_age):
    if pd.isnull(string_age):
        return None
    else:
        new_string = (string_age.split(" "))[0]
        if (new_string[-len(new_string)] == "+"):
            return int(new_string.split("+")[0])
        else:
            return  int(new_string[0:2])
data["int_age"] = (data["Age"].apply(age_conversion))   
print(data["int_age"])
print(data["int_age"].describe())
print(data["Age"].value_counts())

0       18.0
1       18.0
2       18.0
3       30.0
4       30.0
5       18.0
6       18.0
7       18.0
8       30.0
9       30.0
11      30.0
12      18.0
13      18.0
14      60.0
15      30.0
16      30.0
17      18.0
18      30.0
19      30.0
20      30.0
21      18.0
23      18.0
24      30.0
25      18.0
26      30.0
27      18.0
28      30.0
29      30.0
30      30.0
32      18.0
        ... 
1024    45.0
1025    60.0
1026    45.0
1027    60.0
1029    60.0
1030    45.0
1031    45.0
1033    30.0
1034    30.0
1035    30.0
1037    30.0
1038    60.0
1039    18.0
1040    30.0
1041    45.0
1042    45.0
1043    45.0
1044    60.0
1045    30.0
1046    18.0
1047    30.0
1048    60.0
1049     NaN
1050    60.0
1051    60.0
1053    30.0
1054    60.0
1055    60.0
1056     NaN
1057     NaN
Name: int_age, dtype: float64
count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: int_age, dt

# Findings and Analysis

Even though we can see that the age groups are more or less evenly distributed, the fact that we have assigned every age corresponding to a certain range to the lower bound has caused the mean to be lower than it shuold be.

Also, we should keep in mind that some values are NaN, which makes some of the findings somewhat less reliable. The question remains on what we should do with these missing values, drop these rows completely or assign an "average" value?

# Converting Income To Numeric

In [139]:
def income_earned(income):
    if pd.isnull(income):
        return None
    else:
        first_item= income.split(" ")
        if (first_item[0]=="Prefer"):
            return None
        else :
            without_dollar = first_item[0].replace("$", "")
            without_dollar_and_comma = without_dollar.replace(",", "")
            return int(without_dollar_and_comma)
data["int_income"] = data["How much total combined money did all members of your HOUSEHOLD earn last year?"].apply(income_earned)
#print(data["int_income"])  
print(data["int_income"].describe())
data["How much total combined money did all members of your HOUSEHOLD earn last year?"].value_counts()



count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%       25000.000000
50%       75000.000000
75%      100000.000000
max      200000.000000
Name: int_income, dtype: float64


$25,000 to $49,999      166
$75,000 to $99,999      127
$50,000 to $74,999      127
Prefer not to answer    118
$100,000 to $124,999    109
$200,000 and up          76
$10,000 to $24,999       60
$0 to $9,999             52
$125,000 to $149,999     48
$150,000 to $174,999     38
$175,000 to $199,999     26
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: int64

# Findings and Analysis

Similarly to the age, the mean is skewed downwards due to our assigning all income to the lowest bound value of the range. Other than that, it is a true depiction of the incomes of survey participants. 

It is worth to note, nevertheless, that the average is relatively large when the highest income group ($25,000  to $49,999), considerably surpasses the other groups, but that can be explained by the high standard deviation.

# Correlating Travel Distance And Income

In [151]:
data_income_lt_150000 = data [ data ["int_income"] <150000 ]
print ("For people earning less than $150,000:")
print("\n", data_income_lt_150000 [ "How far will you travel for Thanksgiving?" ].value_counts())
print("\n", "In Percentage (%):")
print("\n", data_income_lt_150000 [ "How far will you travel for Thanksgiving?" ].value_counts()/len(data_income_lt_150000)*100)

print ("\nFor people earning more than $150,000:")
data_income_gt_150000 = data [ data ["int_income"] >150000 ]
print("\n", data_income_gt_150000 [ "How far will you travel for Thanksgiving?" ].value_counts())
print("\n", "In Percentage (%):")
print("\n", data_income_gt_150000 [ "How far will you travel for Thanksgiving?" ].value_counts()/len(data_income_gt_150000)*100)



print("\n", "By income groups: \n", data["int_income"].value_counts())

For people earning less than $150,000:

 Thanksgiving is happening at my home--I won't travel at all                         281
Thanksgiving is local--it will take place in the town I live in                     203
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    150
Thanksgiving is out of town and far away--I have to drive several hours or fly       55
Name: How far will you travel for Thanksgiving?, dtype: int64

 In Percentage (%):

 Thanksgiving is happening at my home--I won't travel at all                         40.783745
Thanksgiving is local--it will take place in the town I live in                     29.462990
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    21.770682
Thanksgiving is out of town and far away--I have to drive several hours or fly       7.982583
Name: How far will you travel for Thanksgiving?, dtype: float64

For people earning more than $150,000:

 Thanksgiving is happening at my home--I 

# Findings and Analysis

We see that for people earning less than 150,000 dollars, around 41 percent are not willing to travel at all, while 48 percent of people earning more than 150,000 dollars prefer not to move. Thus, it is reasonable to believe that people earning more than 150,000 dollars are more prone to host Thanksgiving at their residence than those earning less. 

Additionaly, those earning less are more willing to travel for a few hours than those earning more (if Thanksgiving is out of town but not too far). 

However, if Thansgiving is far away, those earning less are less prone to travel than those earning more, likely due to moving and traveling costs.

The fact that people earning less than 150,000 dollars are not very prone to host Thanksgiving but are willing to travel only if Thanksgiving isn't located too far seems to suggest that these people tend to be young people traveling to their hometown to celebrate Thanksgiving with their families. Conversely,the people who earn more and yet are less likely to travel small distances seem to fit the profile of parents who host Thanksgiving in their house.

In order to confirm that the Thanksgiving hosts are more likely to be parents and that those traveling are more likely to be young people traveling to their hometown, it'd be interesting to analyze the relationship between the "Age" column and the "int_income" column.

# Linking Friendship And Age

In [155]:
data.pivot_table(values = "int_age",index = "Have you ever tried to meet up with hometown friends on Thanksgiving night?", columns = 'Have you ever attended a "Friendsgiving?"')

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [156]:
data.pivot_table(values = "int_income",index = "Have you ever tried to meet up with hometown friends on Thanksgiving night?", columns = 'Have you ever attended a "Friendsgiving?"')

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

# Findings and Analysis

In the first pivot table, we notice that people who don't attend Friendsgiving are on average 41-42 years old, an average much larger that that of people who do attend Friendsgiving. 

Looking at the index "Have you ever tried to meet up with hometown friends on Thanksgiving night?", we can already assume that the people who responded "Yes" to that question either live in their hometown or have travelled back home. We observe that the people who celebrate Thanksgiving and meet up hometown friends are much younger than those who meet up with homewtown friends but don't celebrate Friendsgiving. This suggests that younger people are more prone to celebrate Friendsgiving than older people (such as parents), regardless of the actual location of Friendsgiving. This thus explains why the people who don't celebrate Friendsgiving, whether they meet up with hometown friends or not, are around the same age group. Additionally, the analysis also suggests that any activity related to meeting friends (whether it is Friendsgiving or meeting up hometown friends), is usually performed by younger people, while older people's Thanksgiving revolves more around family members.

We can also observe a correlation between the age of the participants and their income. The younger they are, the less income they earn. Thus, the older age group, conveniently the higher income group, neither celebrate Friendsgiving nor meet up with friends during Thanksgiving, reinforcing the hypothesis that those people already live in their hometown and host Thanksgiving themselves. Conversely, the younger people, and those who earn less, are much more likely to celebrate Friendsgiving and travel back to their hometown to celebrate Thanksgiving with their families.

# Identify How Many People Shop on Black Friday

In [168]:
data.pivot_table(values= "int_age", index= "What is your gender?", columns="What is typically the main dish at your Thanksgiving dinner?")

What is typically the main dish at your Thanksgiving dinner?  Chicken  \
What is your gender?                                                    
Female                                                           38.0   
Male                                                             41.0   

What is typically the main dish at your Thanksgiving dinner?   Ham/Pork  \
What is your gender?                                                      
Female                                                        35.600000   
Male                                                          31.846154   

What is typically the main dish at your Thanksgiving dinner?  I don't know  \
What is your gender?                                                         
Female                                                                18.0   
Male                                                                  24.0   

What is typically the main dish at your Thanksgiving dinner?  Other (please specify)  \
What is your gender?                                                                   
Female                                                                     40.500000   
Male                                                                       47.307692   

What is typically the main dish at your Thanksgiving dinner?  Roast beef  \
What is your gender?                                                       
Female                                                              45.0   
Male                                                                36.5   

What is typically the main dish at your Thanksgiving dinner?   Tofurkey  \
What is your gender?                                                      
Female                                                        35.538462   
Male                                                          26.571429   

What is typically the main dish at your Thanksgiving dinner?  Turducken  \
What is your gender?                                                      
Female                                                             60.0   
Male                                                               30.0   

What is typically the main dish at your Thanksgiving dinner?     Turkey  
What is your gender?                                                     
Female                                                        40.453540  
Male                                                          40.472585

In [169]:
data.pivot_table(values= "int_income", index= "What is your gender?", columns="What is typically the main dish at your Thanksgiving dinner?")

What is typically the main dish at your Thanksgiving dinner?  Chicken  \
What is your gender?                                                    
Female                                                        11250.0   
Male                                                          60000.0   

What is typically the main dish at your Thanksgiving dinner?      Ham/Pork  \
What is your gender?                                                         
Female                                                        73571.428571   
Male                                                          56538.461538   

What is typically the main dish at your Thanksgiving dinner?  I don't know  \
What is your gender?                                                         
Female                                                             50000.0   
Male                                                                   0.0   

What is typically the main dish at your Thanksgiving dinner?  Other (please specify)  \
What is your gender?                                                                   
Female                                                                  55250.000000   
Male                                                                   122727.272727   

What is typically the main dish at your Thanksgiving dinner?  Roast beef  \
What is your gender?                                                       
Female                                                           30000.0   
Male                                                             37500.0   

What is typically the main dish at your Thanksgiving dinner?      Tofurkey  \
What is your gender?                                                         
Female                                                        82083.333333   
Male                                                          52000.000000   

What is typically the main dish at your Thanksgiving dinner?  Turducken  \
What is your gender?                                                      
Female                                                         200000.0   
Male                                                           200000.0   

What is typically the main dish at your Thanksgiving dinner?        Turkey  
What is your gender?                                                        
Female                                                        75281.329923  
Male                                                          79220.588235

# Finding Regional Patterns in Main Dishes

In [250]:
unique_region=[]
print(data["US Region"].value_counts())
for i in range(0, 981):
    try:
        if (data["US Region"][i] not in unique_region):
            unique_region.append(data["US Region"][i])
    except KeyError:
        pass


for i in range (0, len(unique_region)):
    region = data [ data ["US Region"] == unique_region[i]]
    print ("\nFor region ", unique_region[i], "\n",region["What is typically the main dish at your Thanksgiving dinner?"].value_counts()/len(region))
    #print(region)



South Atlantic        203
East North Central    145
Middle Atlantic       145
Pacific               130
West South Central     85
West North Central     71
East South Central     56
New England            55
Mountain               41
Name: US Region, dtype: int64

For region  Middle Atlantic 
 Turkey                    0.896552
Tofurkey                  0.034483
Other (please specify)    0.027586
Ham/Pork                  0.013793
Roast beef                0.013793
Turducken                 0.006897
Chicken                   0.006897
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: float64

For region  East South Central 
 Turkey                    0.892857
Other (please specify)    0.071429
Ham/Pork                  0.017857
Roast beef                0.017857
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: float64

For region  Mountain 
 Turkey      0.902439
Tofurkey    0.048780
Chicken     0.024390
Ham/Pork    0.024390
Name: What i